stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -221,7 +241,490 @@ public static ErrorValue createErrorCause(String message, String reason, BPackag
+            }
+            return new WebSocketService((ObjectValue) clientService, strand.scheduler);
+        } else {
+            return new WebSocketService(strand.scheduler);
+        }
+    }
+
+    public static void initialiseWebSocketConnection(String remoteUrl, ObjectValue webSocketClient,
+                                                     WebSocketService wsService) {
+        @SuppressWarnings(WebSocketConstants.UNCHECKED)
+        MapValue<String, Object> clientEndpointConfig = (MapValue<String, Object>) webSocketClient.getMapValue(
+                HttpConstants.CLIENT_ENDPOINT_CONFIG);
+        WebSocketClientConnectorConfig clientConnectorConfig = new WebSocketClientConnectorConfig(remoteUrl);
+        String scheme = URI.create(remoteUrl).getScheme();
+        populateClientConnectorConfig(clientEndpointConfig, clientConnectorConfig, scheme);
+        // Get the connector factory from the native data
+        HttpWsConnectorFactory connectorFactory = ((HttpWsConnectorFactory) webSocketClient.
+                getNativeData(WebSocketConstants.CONNECTOR_FACTORY));
+        // Create the client connector
+        WebSocketClientConnector clientConnector = connectorFactory.createWsClientConnector(clientConnectorConfig);
+        WebSocketClientListenerImpl clientConnectorListener = new WebSocketClientListenerImpl();
+        // Add the client connector as the native data, when client is not as a failover client
+        // Because Here, using one url So no need to create the client connector again
+        webSocketClient.addNativeData(WebSocketConstants.CLIENT_CONNECTOR, clientConnector);
+        webSocketClient.addNativeData(WebSocketConstants.CLIENT_LISTENER, clientConnectorListener);
+        establishInitialWebSocketConnection(webSocketClient, wsService);
     }
 
+    public static WebSocketClientConnector getWebSocketClientConnector(String remoteUrl,
+                                                                       ObjectValue webSocketClient) {
+        @SuppressWarnings(WebSocketConstants.UNCHECKED)
+        MapValue<String, Object> clientEndpointConfig = (MapValue<String, Object>) webSocketClient.getMapValue(
+                HttpConstants.CLIENT_ENDPOINT_CONFIG);
+        WebSocketClientConnectorConfig clientConnectorConfig = new WebSocketClientConnectorConfig(remoteUrl);
+        populateClientConnectorConfig(clientEndpointConfig, clientConnectorConfig, remoteUrl);
+        // Get the connector factory from the native data
+        HttpWsConnectorFactory connectorFactory = ((HttpWsConnectorFactory) webSocketClient.
+                getNativeData(WebSocketConstants.CONNECTOR_FACTORY));
+        // Create the client connector
+        return connectorFactory.createWsClientConnector(clientConnectorConfig);
+    }
+
+    public static void establishInitialFailoverConnection(WebSocketClientConnector clientConnector,
```

Comment:
Same as for constants if the function is used in only one location we can have the function in that class itself instead of creating it in the Util class. Util and Constants classes are for reusability. Shall we check other functions as well?